File: leetcode/1358. Number of Substrings Containing All Three Characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int n = s.size(), ans = 0;
        int lastSeen[3] = {-1, -1, -1};
        for (int i = 0; i < n; i++) {
            lastSeen[s[i] - 'a'] = i;
            if (lastSeen[0] != -1 || lastSeen[1] != -1 || lastSeen[2] != -1) {
                ans = ans + 1 + min({lastSeen[0], lastSeen[1], lastSeen[2]});
            }
        }
        return ans;
    }
};

File: leetcode/1358. Number of Substrings Containing All Three Characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        n, ans = len(s), 0
        last_seen = [-1, -1, -1]
        for i in range(n):
            a = ord(s[i]) - ord('a')
            last_seen[a] = i
            if last_seen[0] != -1 or last_seen[1] != -1 or last_seen[2] != -1:
                ans = ans + 1 + min(last_seen[0], last_seen[1], last_seen[2])
        return ans
        